### ISSUE TITLE
Implement SSH key management for user and portal keys

### ISSUE THREAD

**Issue by ibaldin** [fabric-testbed/UserInformationService#10]
Implement the necessary endpoints and access controls for managing user and portal SSH key(pair)(s)

**Comment by ibaldin** [comment]
Addressed on #25 

In order to migrate the database schema you need to alter fabric_people table:
```
postgres=# alter table fabric_people add column bastion_login character varying;
postgres=# alter table fabric_people add constraint uniq_uuid UNIQUE (uuid);
```

**Comment by ibaldin** [comment]
There are basically 6 new API endpoints. Before I get into what they are an explanation of various field conventions

If you’ve seen what an public key looks like, it typically has 
```
ssh-rsa AAA<a bunch of characters> username@host
```
type structure. Because we want to stay compatible with how Comanage stores keys, in the above structure the first part is referred to as ‘name’ (`ssh-rsa` or `ssh-ecdsa` or other). It is actually more of a key type, but they call it a name. The middle part is base64-encoded public key (referred to as ‘public_key’ in various structures below) and the last part is called ‘comment’. 

GET /keylist – it is for bastion hosts to call to retrieve changed keys (added or expired/deactivated).

GET /sshkeys/{keytype}/{uuid} – this one gets a list of available keys of type ‘bastion’ or ‘sliver’ for user with uuid ‘uuid’. What is returned is a list of SshKeyShort structures (you should be able to see in Swagger. The only thing is ‘public_key’ field will not be returned in this case. Only currently active keys are returned.

POST /sshkeys/{keytype}/{uuid} – upload a public key (entire public key as a single string in standard OpenSSH format) to UIS. UIS internally will break it up into ‘name’, ‘public_key’ and ‘comment’ fields and validate that the key is either RSA of 3072 bits or longer or ECDSA of 256 bits or longer. Other key types (DSA) or shorter lengths will be rejected with an error.

PUT  /sshkeys/{keytype}/{uuid} – generates and returns a keypair as two separate fields ‘private_openssh’ (in PEM format) and ‘public_openssh’ in the standard OpenSSH format where name, public_key and comment are concatenated over spaces. These can basically be saved into files as is (one as private key, other as public) by the user. In the background UIS saves the public key. 

GET /sshkeys/{keytype}/{uuid}/{keyid} – retrieves a full entry of the key including the full public_key string, fingerprint, its status (active or not) and reason for deactivation (auto-expired or removed by user using DELETE call below). 

DELETE /sshkeys/{keytype}/{uuid}/{keyid} – deactivate a given key – doesn’t actualy delete, but it will no longer be returned by the GET /sshkeys/{keytype}/{uuid}. It can still be retrieved by the GET /sshkeys/{keytype}/{uuid}/{keyid} but it wil be marked as deactivated. 

In this version keys expire after 5 minutes and are permanently deleted after 10 minutes. So upon upload the key is active, 5 minutes later it because inactive (auto-expires) and after 5 more minutes gets permanently deleted.

**Comment by ibaldin** [comment]
Addressed on #25
